### PR TITLE
N8n 2283 implement design system

### PR DESCRIPTION
**Pull request by mutdmour**
- implement n8n design system, can be previewed here https://n8n-io.github.io/n8n-design-system/index.html
- move all el-button elements to the new components
- break apart element io imports, removing unused components, reducing overall js by 10%
- replace element io scss theme with n8n custom n8n css theme
- update select elements to match button sizes

### CODE REVIEW

**Review by mutdmour — COMMENTED** [review]
hints

File: packages/editor-ui/package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "test:unit": "vue-cli-service test:unit"
   },
   "dependencies": {
+    "n8n-design-system": "git+https://github.com/n8n-io/n8n-design-system.git",
```
Comment (mutdmour) [inline packages/editor-ui/package.json:28]: todo - change to package version number once published to npm

**Review by mutdmour — COMMENTED** [review]
hints

File: packages/editor-ui/src/components/Modal.vue [hunk 2]
```diff
@@ -87,7 +87,7 @@ export default Vue.extend({
 		},
 		closeDialog(callback?: () => void) {
 			this.$store.commit('ui/closeTopModal');
-			if (callback) {
+			if (typeof callback === 'function') {
```
Comment (mutdmour) [inline packages/editor-ui/src/components/Modal.vue:90]: fix console error when clicking on cancel in duplicate workflow modal

File: packages/design-system/src/main.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import * as components from './components';
+
+for (const key in components) {
+	const component = components[key];
+	component.install = function (Vue) {
+		Vue.component(component.name, component);
+	};
+}
```
Comment (BHesseldieck) [inline packages/design-system/src/main.js:8]: matter of taste, but would prefer a functional approach here

File: packages/design-system/tslint.json [hunk 4]
```diff
@@ -0,0 +1,80 @@
+{
```
Comment (BHesseldieck) [inline packages/design-system/tslint.json:1]: I guess it's just copied, however setting up `tslint` for this package is not worth it and we should directly setup the `eslint` linter 👍🏼 

accordingly wouldn't add it to the github action

File: packages/editor-ui/src/components/SaveWorkflowButton.vue [hunk 5]
```diff
@@ -1,14 +1,10 @@
 <template>
-	<el-button :disabled="isWorkflowSaving" :class="{saved: isSaved}" size="small" @click="save">
-		<font-awesome-icon v-if="isWorkflowSaving" icon="spinner" spin />
-		<span v-else-if="isDirty || isNewWorkflow">
-			Save
-		</span>
-		<span v-else>Saved</span>
-	</el-button>
+	<span :class="$style.container">
+		<n8n-button label="Save" :disabled="isWorkflowSaving" @click="save" v-if="isDirty || isNewWorkflow" />
```
Comment (BHesseldieck) [inline packages/editor-ui/src/components/SaveWorkflowButton.vue:3]: let's have the `v-if` as first attribute

File: packages/editor-ui/src/components/RunData.vue [hunk 6]
```diff
@@ -55,27 +53,24 @@
...
-			<div v-if="node && workflowRunData !== null && workflowRunData.hasOwnProperty(node.name) && !workflowRunData[node.name][runIndex].error" class="title-data-display-selector" @click.stop>
+			<div v-if="hasNodeRun && !hasRunError" class="title-data-display-selector" @click.stop>
 				<el-radio-group v-model="displayMode" size="mini">
 					<el-radio-button label="JSON" :disabled="showData === false"></el-radio-button>
 					<el-radio-button label="Table"></el-radio-button>
 					<el-radio-button label="Binary" v-if="binaryData.length !== 0"></el-radio-button>
 				</el-radio-group>
 			</div>
-			<div class="select-button" v-if="displayMode === 'JSON' && state.path !== deselectedPlaceholder">
+			<div class="select-button" v-if="hasNodeRun && !hasRunError && displayMode === 'JSON' && state.path !== deselectedPlaceholder">
```
[the remark this answers is not in the document]
Comment (BHesseldieck) [reply packages/editor-ui/src/components/RunData.vue:63]: let's have the `v-if` as first attribute

**Review by BHesseldieck — APPROVED** [review]
tremendous work 🏅 🚀 

love it!

just saw 14 empty files and wanted to check that we remove them if they are not required 👍🏼 
examples:
`packages/design-system/theme/src/breadcrumb-item.scss`
`packages/design-system/theme/src/checkbox-button.scss`
`packages/design-system/theme/src/checkbox-group.scss`
`packages/design-system/theme/src/collapse-item.scss`

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with n8n@0.136.0

**Pull request merged**